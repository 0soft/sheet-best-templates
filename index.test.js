global.fetch = require('isomorphic-fetch');
const SheeBest = require('./index');

const sleep = (seconds) => new Promise((resolve) => setTimeout(resolve, seconds * 1000));

test('Reading a SpreadSheet and turning a template into formatted HTML', async (done) => {
  jest.setTimeout(30000);

  document.body.innerHTML = `
<div data-sheet-best="https://sheet.best/api/sheet/cf969697-682a-40e3-bad4-d54803eeeacf">
  <div>
    <h2>{{ code }}</h2>
    <div>
      <span>Price: <b>{{ price }}</b></span>
      <span>High: <b>{{ high }}</b></span>
      <span>Low: <b>{{ low }}</b></span>
    </div>
  </div>
</div>
`.trim();

  await SheeBest.setup();
  expect(document.body.innerHTML).toMatchSnapshot();

  done();
});

test('Writing data to a SpreadSheet', async (done) => {
  global.fetch = jest.fn(() => Promise.resolve({
    json: () => ({}),
  }));
  global.FormData = jest.fn(() => () => ({}));
  Array.from = jest.fn(() => [
    ['code', 'FOOO'], ['price', '2000'], ['volume', '10000000'],
  ]);

  document.body.innerHTML = `
<form id="form" data-sheet-best="https://sheet.best/api/sheet/cf969697-682a-40e3-bad4-d54803eeeacf">
  <input type="text" name="code"/>
  <input type="text" name="price"/>
  <input type="text" name="volume"/>
  <button id="submit-button" type="submit">submit</button>
</form>
`.trim();

  await SheeBest.setup();

  const button = document.getElementById('submit-button');
  button.click();

  await sleep(1.4);

  expect(fetch).toHaveBeenCalledTimes(1);

  done();
});
