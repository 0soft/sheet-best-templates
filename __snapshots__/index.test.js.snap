// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reading a SpreadSheet and turning a template into formatted HTML 1`] = `
"<div data-sheet-best=\\"https://sheet.best/api/sheet/cf969697-682a-40e3-bad4-d54803eeeacf\\">
  <div>
    <h2>COST</h2>
    <div>
      <span>Price: <b>273.59</b></span>
      <span>High: <b>275.36</b></span>
      <span>Low: <b>272.76</b></span>
    </div>
  </div>

  <div>
    <h2>AAPL</h2>
    <div>
      <span>Price: <b>205.22</b></span>
      <span>High: <b>206.44</b></span>
      <span>Low: <b>202.7</b></span>
    </div>
  </div>

  <div>
    <h2>ADBE</h2>
    <div>
      <span>Price: <b>286.59</b></span>
      <span>High: <b>289.65</b></span>
      <span>Low: <b>286</b></span>
    </div>
  </div>

  <div>
    <h2>AMZN</h2>
    <div>
      <span>Price: <b>1779.99</b></span>
      <span>High: <b>1795.65</b></span>
      <span>Low: <b>1774.61</b></span>
    </div>
  </div>

  <div>
    <h2>PEP</h2>
    <div>
      <span>Price: <b>129.8</b></span>
      <span>High: <b>130.39</b></span>
      <span>Low: <b>129.18</b></span>
    </div>
  </div>

  <div>
    <h2>TSLA</h2>
    <div>
      <span>Price: <b>229.31</b></span>
      <span>High: <b>231.5</b></span>
      <span>Low: <b>228.03</b></span>
    </div>
  </div>
</div>"
`;
